# This file was adapted from the dbt-core repository. The original can
# be found here:
#
#     https://github.com/dbt-labs/dbt-core/blob/48464a22a48e9973745faf30acb8faf0a3dcd65e/.github/ISSUE_TEMPLATE/bug-report.yml
#
# The original is licensed under the Apache v2.0 License:
#
#     Copyright 2021 dbt Labs, Inc.
#
#     Licensed under the Apache License, Version 2.0 (the "License");
#     you may not use this file except in compliance with the License.
#     You may obtain a copy of the License at
#
#         http://www.apache.org/licenses/LICENSE-2.0
#
#     Unless required by applicable law or agreed to in writing, software
#     distributed under the License is distributed on an "AS IS" BASIS,
#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#     See the License for the specific language governing permissions and
#     limitations under the License.
#

name: 🐞 Bug
description: Report a bug or an issue you've found with aiootp
title: "fix(category-or-scope): lowercase-title [#related-github-id][#...]"
labels: ["bug"]
body:
  - type: markdown
    attributes:
      value: |
        Thanks for taking the time to fill out this bug report!
  - type: checkboxes
    attributes:
      label: Is this a new bug in aiootp?
      description: >
        In other words, is this a new error, flaw, failure, or fault in our software?

        Is this a security issue? Follow the instructions in our [Security Report Document](https://github.com/rmlibre/aiootp/blob/main/SECURITY.md).

        Is this bug related to a supported version of aiootp? Check the [Supported Versions](https://github.com/rmlibre/aiootp/blob/main/SECURITY.md) & only file this report if it is.

        Is this a request for help or troubleshooting code in your own project? Open a [Discussion Question](https://github.com/rmlibre/aiootp/discussions).

        Please search to see if an issue already exists for the bug you encountered.
      options:
        - label: This affects a supported version of aiootp
          required: true
        - label: This isn't a security vulnerability or issue
          required: true
        - label: I've searched the existing issues, & I couldn't find one for this bug
          required: true
  - type: textarea
    attributes:
      label: Current Behavior
      description: A concise description of what you're experiencing.
    validations:
      required: true
  - type: textarea
    attributes:
      label: Expected Behavior
      description: A concise description of what you expected to happen.
    validations:
      required: true
  - type: textarea
    attributes:
      label: Steps To Reproduce
      description: Steps to reproduce the behavior.
      placeholder: |
        1. In this environment ___
        2. With this config ___
        3. Run ___
        4. See error ___
    validations:
      required: true
  - type: textarea
    id: logs
    attributes:
      label: Relevant log output
      description: |
        If applicable, log output to help explain your problem.
      render: shell
    validations:
      required: false
  - type: textarea
    attributes:
      label: Environment
      description: |
        examples:
          - **OS**: Ubuntu 22.04
          - **Python**: 3.8.10 (`python3 --version`)
          - **aiootp**: 0.23.7 (`import aiootp; aiootp.__version__`)
      value: |
        - OS: ___
        - Python: ___
        - aiootp: ___
      render: markdown
    validations:
      required: false
  - type: textarea
    attributes:
      label: Additional Context
      description: |
        Links? References? Fix suggestions? Anything that'll help us resolve the issue!

        Tip: You can attach images or log files by clicking this area to highlight it & then dragging files in.
    validations:
      required: false
